Mathlib/Data/Finsupp/Weight.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.Algebra.Order.BigOperators.Group.Finset
+import Mathlib.Algebra.Order.Module.Defs
+import Mathlib.Algebra.Order.Monoid.Canonical.Defs
+import Mathlib.LinearAlgebra.Finsupp
+
+
+/-! # weights of Finsupp functions -/
+
+variable {σ M : Type*} (w : σ → M)
+
+namespace Finsupp
+
+section AddCommMonoid
+
+variable [AddCommMonoid M]
+/-- The `weight` of the finitely supported function `s : σ →₀ ℕ`
+with respect to `w : σ → M` is the sum `∑(s i)•(w i)`. -/
+noncomputable def weight : (σ →₀ ℕ) →+ M :=
+  (Finsupp.total σ M ℕ w).toAddMonoidHom
+
+theorem weight_apply (f : σ →₀ ℕ) :
```

Comment:
I guess `simps` can generate this.

Reply:
It does not generate the same result, it is nice to go way down to `Finsupp.sum`